### DISCUSSION TITLE
development status table and map of methods

### DISCUSSION THREAD

**Discussion by ken-maeda** [TDAmeritrade/stumpy#702] [General]
I took a while to understand development status and method usage. It might be helpfull if table and map for it are?

Table:
![image](https://user-images.githubusercontent.com/42758592/199202925-3bd6a657-32d7-47c3-8798-f7794ed5f06a.png)

Maps: like scikit learn maps?
![image](https://user-images.githubusercontent.com/42758592/199203126-071bb5cd-9c28-4649-ba7e-912bc95daadb.png)

**Comment by seanlaw** [comment]
@ken-maeda That's a very interesting idea! I really like the idea of having some sort of visual map similar to the sklearn one that gives them a decision-tree like guidance. The table is also valuable as it directly relates the published work directly onto the STUMPY functions.

I don't know if it's too much information but, essentially, for all functions, we have its equivalent `normalize=False` equivalent as well as all possible `p`-norm support as well. I'm not sure how to proceed from here but I'm thinking that we test-and-learn as we go. Maybe we discover that it's too complicated/hard and we can stop OR we iterate and find a good way to do it. How does that sound to you?

I know that @NimaSarajpoor is a big sklearn proponent and so I'd love to get his thoughts too.

**Reply by ken-maeda** [reply]
I agree with you. I'm not sure which way is best for sharing such spread sheet. I'll try to find it.
About the map, by user dataset and the purpose, map should pick out functions worth trying from stumpy at least. (at first)

**Reply by NimaSarajpoor** [reply]
@ken-maeda  @seanlaw 
I think this is a very nice idea. I had a similar idea when my friend introduced the tool "Obsidian" to me and  I was thinking of creating something similar to [Mind Maps](https://iwannabemewhenigrowup.medium.com/using-mind-maps-with-obsidian-bb2fdfdda0cd) for STUMPY. But, I then thought it could get messy. 

The scikit-learn's example is nice though. The good thing about this scikit-learn map is that it is clear and well categorized. I do not know how it will look like for STUMPY...but I feel it would be useful if we can create a nice visual map (because, as I said, I had a similar idea. I am not sure why I had such idea but I think the reason was that I got a little bit confused when I was working on TopK PR, and I thought it would be nice if we have a visual map that can guide reader through different capabilities of STUMPY)

So, to sum up... I support @ken-maeda's idea as I think this scikit-learn visual map can be useful :) Also, we may add link to the nodes of the decision-tree like map so it can take user to the relevant tutorial section!

**Reply by seanlaw** [reply]
@ken-maeda If you'd like to spend some time on it, I support you and happy to provide feedback!
